openstef_dbc/services/weather.py

```diff
@@ -97,22 +97,19 @@ def _get_nearest_weather_location(
                 )
 
             distance = round(
-                geopy.distance.geodesic(coordinates, location_coordinates).km
+                geopy.distance.geodesic(coordinates, location_coordinates).km, 1
             )
             city = weather_location["city"]
             distance_df = pd.DataFrame([{"distance": distance, "input_city": city}])
             distances = pd.concat([distances, distance_df])
 
         distances = distances.set_index("distance")
 
-        nearest_location = distances["input_city"].sort_index()[0:k].tolist()
+        nearest_location = distances["input_city"].sort_index().iloc[0:number_locations]
```

Comment:
You may want to use nsmallest
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.nsmallest.html
```suggestion
        nearest_location = distances.nsmallest(number_locations, "input_city")
```

Reply:
Unfortunately, no ... I want to extract the name of the input_city that have the smallest distances.
Therefore i sort the index (which contains the distance) and keep only "input_city" columns n first elements.
nsmallest does not allow to sort with "index". 
With your suggestion, i get the following error message :
`TypeError: Column 'input_city' has dtype object, cannot use method 'nsmallest' with this dtype`

Reply:
Is it ok to resolve this ?

Reply:
Yes, sorry for the late answer! I see what you're trying to do, shame it doesn't work. Looks good to me then!